src/vm/jitinterface.cpp

```diff
@@ -9105,30 +9105,47 @@ CorInfoType CEEInfo::getFieldTypeInternal (CORINFO_FIELD_HANDLE fieldHnd,
     // <REVISIT_TODO>TODO should not burn the time to do this for anything but Value Classes</REVISIT_TODO>
     _ASSERTE(type != ELEMENT_TYPE_BYREF);
 
-    // For verifying code involving generics, use the class instantiation
-    // of the optional owner (to provide exact, not representative,
-    // type information)
-    SigTypeContext typeContext(field, (TypeHandle) owner);
+    if (type == ELEMENT_TYPE_I)
+    {
+        PTR_MethodTable enclosingMethodTable = field->GetApproxEnclosingMethodTable();
+        if (enclosingMethodTable->IsByRefLike() &&
+            (
+                enclosingMethodTable == g_TypedReferenceMT && field->GetOffset() == 0 ||
+                enclosingMethodTable->HasSameTypeDefAs(g_pByReferenceClass)
+            ))
+        {
+            _ASSERTE(field->GetOffset() == 0);
+            type = ELEMENT_TYPE_BYREF;
```

Comment:
At the top of the function it already does `*pTypeHnd = 0`, so I guess it's not null currently. Should it check for null?